algorithms/src/Kokkos_Random.hpp

```diff
@@ -617,9 +617,10 @@ struct Random_XorShift1024_UseCArrayState<Kokkos::Experimental::OpenMPTarget>
     : std::false_type {};
 #endif
 
-template <class ExecutionSpace>
+template <class ExecutionSpace, class MemorySpace>
 struct Random_UniqueIndex {
```

Comment:
Template it on the device type

Reply:
See [`c7fbcb7` (#4724)](https://github.com/kokkos/kokkos/pull/4724/commits/c7fbcb79cfbad0406200ab7fa89159f11edd8573).